flang/docs/CoarrayFortranRuntime.md

```diff
@@ -7,21 +7,37 @@
-  `sync-team-stmt`, `event-post-stmt`, `event-wait-stmt`,` error-stop-stmt`, `lock-stmt`. The intrinsic functions include
+  In order to be fully Fortran 2018 compliant, Flang needs to add support for what is commonly referred to as coarray fortran, which includes features related to parallelism. These features include the following statements, subroutines, functions, types, and kind type parameters:
+  
+  * **Statements:** 
+    - _Synchronization:_ `sync all`, `sync images`, `sync memory`, `sync team` 
+    - _Events:_ `event post`, `event wait`
+    - _Error termination:_ `error stop`
+    - _Locks:_ `lock`, `unlock` 
+    - _Failed images:_ `fail image`
+    - _Teams:_ `form team`, `change team`
+    - _Critical sections:_ `critical`, `end critical`
+  * **Intrinsic functions:**
   `num_images`, `this_image`, `lcobound`, `ucobound`, `team_number`, `get_team`, `failed_images`, `stopped_images`,
-  `image_status`, `coshape`, `image_index`. The intrinsic subroutines include `event_query`, the collectives: `co_sum`, `co_max`,
-  `co_min`, `co_reduce`, `co_broadcast` and the atomics: `atomic_add`, `atomic_and`, `atomic_cas`, `atomic_define`,
-  `atomic_fetch_add`, `atomic_fetch_and`, `atomic_fetch_or`, `atomic_fetch_xor`, `atomic_or`, `atomic_ref`, `atomic_xor`.
-
+  `image_status`, `coshape`, `image_index`
+  * **Intrinsic subroutines:**
+    - _Collective subroutines:_ `co_sum`, `co_max`, `co_min`, `co_reduce`, `co_broadcast`
+    - _Atomic subroutines:_ `atomic_add`, `atomic_and`, `atomic_cas`, `atomic_define`,
+  `atomic_fetch_add`, `atomic_fetch_and`, `atomic_fetch_or`, `atomic_fetch_xor`, `atomic_or`, `atomic_ref`, `atomic_xor`
+    - _Other subroutines:_ `event_query`
+  * **Types, kind type parameters, and values:**
+    - _Intrinsic derived types:_ `event_type`, `team_type`
+    - _Atomic kind type parameters:_ `atomic_int_kind` and `atomic_logical_kind`
+    - _Values:_ `stat_failed_image`, `stat_locked`, `stat_locked_other_image`, `stat_stopped_image`, `stat_unlocked`, `stat_unlocked_failed_image`
 
+In addition to being able to support syntax related to the above features, compilers will also need to be able to handle new execution concepts such as image control.  The image control concept affects the behaviors of some statements that were introduced in Fortran expressly for supporting parallel programming, but image control also affects the behavior of some statements that pre-existed parallism in standard Fortran:
+ * **Image control statements:**
+   - _Pre-existing statements_: `allocate`, `deallocate`, `stop`, `end`, a `call` referencing `move_alloc` with coarray arguments
+   - _New statements:_ `sync all`, `sync images`, `sync memory`, `sync team`, `change team`, `end team`, `critical`, `end critical`, `event post`, `event wait`, `form team`, `lock`, `unlock`
+, one consequence of which is to limit the code movement in optimizing compilersSupporting some required behavior will necessitate modifying the behavior of existing
```

Comment:
Is this line meant to be here? Please remove or fix to be complete sentences and in the correct place.

Reply:
Good catch, @ktras.  Something got mistakenly deleted before I committed this.  I'll fix it.

Reply:
@ktras commit c842a890b2d9c01bfce09669a8a61c0fc78efdfa addresses this so I'll mark this conversation as resolved.